PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+  "extends": ["@parcel/config-default", "@company/parcel-config"]
+}
+```
+
+When extending a config, Parcel does a shallow merge of the two configs.
+
+#### `.parcelrc#resolvers`
+
+`resolvers` is an array of strings that specifies the name of a Parcel resolver
+package.
+
+```json
+{
+  "resolvers": ["@parcel/resolver-default"]
+}
+```
+
+See [Resolvers](#resolvers)
+
+#### `.parcelrc#transforms`
+
+`transforms` is an object map of globs to arrays of Parcel transform packages.
+
+```json
+{
+  "transforms": {
+    "*.js": ["@parcel/transform-babel"]
+  }
+}
+```
+
+See [Transforms](#transforms)
+
+#### `.parcelrc#bundler`
+
+`bundler` is a string that specifies the name of a Parcel bundler package.
+
+```json
+{
+  "bundler": "@parcel/bundler-v1"
+}
+```
+
+See [Bundlers](#bundlers)
+
+#### `.parcelrc#packagers`
+
+`packagers` is an object map of globs to Parcel packager packages.
+
+```json
+{
+  "packagers": {
+    "*.js": ["@parcel/packager-js"]
+  }
+}
+```
+
+See [Packagers](#packagers)
+
+#### `.parcelrc#optimizers`
+
+`optimizers` is an object map of globs to arrays of Parcel optimizer packages.
+
+```json
+{
+  "optimizers": {
+    "*.js": ["@parcel/optimizer-uglify"]
+  }
+}
+```
+
+See [Optimizers](#optimizers)
+
+#### `.parcelrc#loaders`
+
+`loaders`  is an object map of globs to Parcel loader packages. See
+[Loaders](#).
+
+```json
+{
+  "loaders": {
+    "*.js": "@parcel/loader-js",
+    "*.wasm": "@parcel/loader-wasm"
+  }
+}
+```
+
+See [Loaders](#loaders)
+
+#### `.parcelrc#reporters`
+
+`reporters` is an array of Parcel reporter packages. See [Reporters](#).
+
+```json
+{
+  "reporters": ["@parcel/reporter-detailed"]
+}
+```
+
+See [Reporters](#reporters)
+
+## Parcel Architecture
+
+Even if you aren't doing anything that complex, if you are going to use Parcel
+a lot it makes sense to take some time and understand how it works.
+
+### Phases of Parcel
+
+At a high level Parcel runs through several phases:
+
+- Resolving
+- Transforming
+- Bundling
+- Packaging
+- Optimizing
+
+The **resolving** and **transforming** phases work together in parallel to
+build a graph of all your assets.
+
+This asset graph gets translated into bundles in the **bundling** phase.
+
+Then the **packaging** phase takes the assets in the calculated bundles and
+merges them together into files each containing an entire bundle.
+
+Finally, in the **optimizing** phase, Parcel takes these bundles files and runs
+them through optimizing transforms.
+
+### Asset Graph
+
+During the resolving and transforming phases, Parcel discovers all the assets
+in your app or program. Every asset can have it's own dependencies on other
+assets which Parcel will pull in.
+
+The data structure that represents all of these assets and their dependencies
+on one another is known as "The Asset Graph".
+
+| Asset Name   | Dependencies        |
+| ------------ | ------------------- |
+| `index.html` | `app.css`, `app.js` |
+| `app.css`    | N/A                 |
```

Comment:
what if the css references other dependencies, like background-images or fonts ?

Reply:
This is already happening, but this references are not inline by default. They are, however, bundled and processed.

So I believe this usage is implied, though not explicitly stated.

Reply:
That is already happening, but the references are not inline by default. They are, however, bundled and processed.

So I believe this usage is implied, though not explicitly stated.

Reply:
this is an example